### DISCUSSION TITLE
issues about hydrodyn damping

### DISCUSSION THREAD

**Discussion by XinmengZeng** [OpenFAST/openfast#873] [Q&A]
I have confused about the additional linear stiffness and linear damping and quadratic damping for monopile and floating.
If i have three free decay time history about surge\heave\pitch through expriment or CFD, Using PQ damping method to get P and Q and  equivalent linear damping ratio, how to setting the three matrix?
And if  i did not have the free decay,the three matrix is it for zero?
How about monopile?
And the quadratic damping matrix is it conflict with HYDRODYNAMIC COEFFICIENTS below,and it is must choose one only?

**Comment by jjonkman** [comment]
Dear @XinmengZeng,

The additional linear stiffness (`AddCLin`), linear damping (`AddBLin`), and quadratic drag (`AddBQuad`) matrix inputs in HydroDyn are there for adding stiffness, damping, and drag that may not be modeled directly in the numerical model, but are seen in the validation data (experimental data or results from high fidelity models).  These only need to be set if the underlying physics are not captured directly in other features of the model (e.g., mooring or hydrostatic stiffness, wave-radiation or structural damping, viscous drag, etc.).

When validating your OpenFAST model, I would suggest first setting `AddCLin`, `AddBLin`, and `AddBQuad` to zero, unless you are aware of specific values that need to be specified by known unmodeled physics.  Then, based on the free-decay results and subsequent P-Q analysis, you can see if the natural frequencies are correct (if not, may require setting `AddCLin`), linear damping (P) is correct (if not, may require setting `AddBLin`), and quadratic drag (Q) is correct (if not, may require setting `AddBQuad`).  For uncoupled motions, the P and Q terms can likely be directly related to diagonal elements of the `AddBLin` and `AddBQuad` matrices, but for motions that are coupled (e.g., surge-pitch-heave coupling), then this may not be simple, and may require a manual tuning or an optimization procedure to set.  It may also be preferred to calibrate the physical properties of the model (mooring stiffness, viscous drag coefficients, etc.) rather than setting nonzero values of `AddCLin`, `AddBLin`, and `AddBQuad`.

I hope that helps.

Best regards,

**Comment by XinmengZeng** [comment]
Dear @jjonkman 
      Thanks for your kind help，can i ask for you another question？
      The hydrodynamic coefficient(morison coefficient,cd?) is it the same with quadratic drag? So, if i set the hydrodynamic coefficient, i needn't to setting quadratic drag in fast?

Best wishes

**Comment by jjonkman** [comment]
Dear @XinmengZeng,

The quadratic drag matrix and the hydrodynamic drag coefficients are not the same thing, but they have similarities.  The similarity is that they are both quadratic in nature (proportional to v*ABS(v), where "v" is the velocity).  However, the hydrodynamic drag model in the strip-theory solution is based on the relative velocity between the wave particles and structure and may also consider structural flexibility, whereas the quadratic drag matrix does not consider wave kinematics and only applies to rigid bodies.

Best regards,
